sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernatePoolDao.java

```diff
@@ -1,211 +1,187 @@
-
-  @Autowired
-  private SessionFactory sessionFactory;
-
-  public Criteria createCriteria() {
-    return currentSession().createCriteria(PoolImpl.class);
+  public HibernatePoolDao() {
+    super(Pool.class, PoolImpl.class);
   }
 
-  @Override
-  public Session currentSession() {
-    return sessionFactory.getCurrentSession();
-  }
+  private final static List<SingularAttribute<PoolImpl, String>> IDENTIFIER_PROPERTIES =
+      Arrays.asList(PoolImpl_.name, PoolImpl_.alias, PoolImpl_.identificationBarcode);
 
-  @Override
-  public Pool get(long poolId) throws IOException {
-    return (PoolImpl) currentSession().get(PoolImpl.class, poolId);
-  }
+  private final static List<SingularAttribute<PoolImpl, String>> SEARCH_PROPERTIES =
+      Arrays.asList(PoolImpl_.name, PoolImpl_.alias, PoolImpl_.identificationBarcode, PoolImpl_.description);
 
   @Override
   public Pool getByBarcode(String barcode) throws IOException {
     if (barcode == null)
       throw new NullPointerException("cannot look up null barcode");
-    return (Pool) createCriteria().add(Restrictions.eq("identificationBarcode", barcode)).uniqueResult();
+    return getBy(PoolImpl_.IDENTIFICATION_BARCODE, barcode);
   }
 
   @Override
   public Pool getByAlias(String alias) throws IOException {
-    return (Pool) createCriteria().add(Restrictions.eq("alias", alias)).uniqueResult();
-  }
-
-  public SessionFactory getSessionFactory() {
-    return sessionFactory;
-  }
-
-  @Override
-  public List<Pool> listAll() throws IOException {
-    @SuppressWarnings("unchecked")
-    List<Pool> results = createCriteria().list();
-    return results;
+    return getBy(PoolImpl_.ALIAS, alias);
   }
 
   @Override
   public List<Pool> listByLibraryId(long libraryId) throws IOException {
+    QueryBuilder<?, PoolImpl> builder =
+        new QueryBuilder<>(currentSession(), PoolImpl.class, Object[].class, Criteria.DISTINCT_ROOT_ENTITY);
+    Join<PoolImpl, PoolElement> elementJoin = builder.getJoin(builder.getRoot(), PoolImpl_.poolElements);
+    Join<PoolElement, ListLibraryAliquotView> aliquotJoin = builder.getJoin(elementJoin, PoolElement_.aliquot);
+    Join<ListLibraryAliquotView, ParentLibrary> libraryJoin =
+        builder.getJoin(aliquotJoin, ListLibraryAliquotView_.parentLibrary);
+    builder.addPredicate(builder.getCriteriaBuilder().equal(libraryJoin.get(ParentLibrary_.libraryId), libraryId));
+    builder.setColumns(builder.getRoot());
     @SuppressWarnings("unchecked")
-    List<Pool> results = currentSession().createCriteria(PoolImpl.class)
-        .createAlias("poolElements", "element")
-        .createAlias("element.aliquot", "aliquot")
-        .createAlias("aliquot.parentLibrary", "library")
-        .add(Restrictions.eq("library.id", libraryId))
-        .setResultTransformer(Criteria.DISTINCT_ROOT_ENTITY)
-        .list();
+    List<Pool> results = (List<Pool>) builder.getResultList();
```

Comment:
Why the `Object[].class` parameter? Does this not work properly with `Pool.class` and removing the `setColumns` call and the cast on `results`?